### ISSUE TITLE
Bug: Cannot insert a destroyed View into a ViewContainer

### ISSUE THREAD

**Issue by lansana** [angular/angular#22429]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior

I am using `ContentChild` and `ng-template` to use named templates in the parent component, so they can be rendered in the child component. When some logic changes (causing change detection to run), I get `Cannot insert a destroyed View into a ViewContainer` error.

## Expected behavior

The desired behavior is the `ContentChild` and `ng-template` logic should stay consistent and always work regardless of what values change and when change detection runs.

## Minimal reproduction of the problem with instructions
You can see the component/template/usage here on my StackOverflow question: https://stackoverflow.com/questions/48937362/angular-5-cannot-insert-a-destroyed-view-into-a-viewcontainer

## What is the motivation / use case for changing the behavior?
My use case is to create a component that wraps the logic of pagination while letting me use named templates such that I can be as expressive as I want and do things like:

- If there are no results, render this content in the #noResults template
- If there are results, render this content in the #results template

...so on and so forth. This prevents me from having copy/pasted code in every component that I want to have pagination logic in, because it gets hard to maintain.

## Environment

<pre><code>
Angular version: 5.2.6

Browser:
- [x] Chrome (desktop) version 59.0.3071.109
 
For Tooling issues:
- Node version: 8.7.0

**Comment by lansana** [comment]
Fixed this. The explanation is on my answer on the StackOverflow question: https://stackoverflow.com/a/48978368/4749297

TL;DR: I was subscribing to an observable in a directive, and never unsubscribed in the `ngOnDestroy` hook. When the event was triggered, the directive was trying to insert a component and it resulted in the error I received.

**Comment by lansana** [comment]
Just to add a quick note, the reason this even required a ticket was because I got no stack trace. For some reason, I just got a message passed into my `ToastService` function (I assume this is because the error was somehow passed into the stream of that `ToastService` observable). This silenced the stack trace and I was left with just a UI toast message saying "Cannot insert a destroyed View into a ViewContainer", so this made it increasingly difficult to debug. If I had a stack trace and could see the last function my app called, problem would've been solved quickly.

**Comment by beeman** [comment]
I got this error when dynamically rendering a [formly])(https://github.com/formly-js/ngx-formly/) form in an Angular Material dialog. I did not find the root cause but it got fixed by using [cloneDeep](https://lodash.com/docs/4.17.11#cloneDeep) when passing in my fields
